### DISCUSSION TITLE
hwdec=vulkan-copy

### DISCUSSION THREAD

**Discussion by ghost** [hooke007/mpv_PlayKit#280] [mpv 通用]

**Comment by skiqing** [comment]
这么牛，能把包放出来吗？

**Reply by skiqing** [reply]
T.T 不会，我U是12代的应该能跑得动

**Reply by skiqing** [reply]
直接覆盖MPV.exe吗？还要改哪里？覆盖完还是不显示vulkan-copy，我显卡6800xt

**Reply by skiqing** [reply]
已尝试mpv.conf里加上hwdec=vulkan-copy，没效果。vulkaninfo.exe里没有那些扩展。。。人麻了。。。垃圾AMD

**Reply by skiqing** [reply]
vulkaninfo.exe里倒是有了，但是用你的包一播放程序就闪退

**Reply by skiqing** [reply]
AMD驱动的问题。。垃圾amd

**Reply by skiqing** [reply]
是AMD驱动的原因，AMD内核和AMD面板不兼容有BUG，和程序无关

**Comment by hooke007** [comment]
从这个pr https://github.com/mpv-player/mpv/pull/9975 知道的vk的解码，对我来说更大的意义是带来大量硬件滤镜的支持

**Comment by ghost** [comment]


**Reply by hooke007** [reply]
你那边d3d11va-copy比较好吗，这里一直都是远不如dxva2-copy的

**Comment by ghost** [comment]


**Reply by hooke007** [reply]
。。。文艺复兴（

**Comment by hingbong** [comment]
AMD 5800U的核显+最新驱动，没有那几个扩展，看来是不支持了。
我在这里编译了集成这个ffmpeg的mpv
[hingbong/mpv-winbuild](https://github.com/hingbong/mpv-winbuild/actions/runs/4691884289)

**Comment by ausar47** [comment]
请问vulkan和vulkan-copy哪个更好呢？渲染还是用d3d11吗？

**Reply by ausar47** [reply]
> > 请问vulkan和vulkan-copy哪个更好呢？
> 
> 没有cpu滤镜需求的话当然是vulkan更好，vulkan-copy主要是相比其它copy hwdec，性能有巨大提升(我设备上200%+)，对于thumbfast之类的场景很有用
> 
> > 渲染还是用d3d11吗？
> 
> vulkan解码必须vulkan渲染，vulkan-copy已经hwdownload了，没有任何要求

vulkan-copy确实牛逼，实测下来比d3d11va还强
不过我用了你上面编译好的mpv好像只有vulkan-copy没有vulkan，设置hwdec=vulkan gpu-next=winvk打开视频没有显示hwdec=vulkan

**Comment by zxhzxhz** [comment]
![Snipaste_2023-04-24_22-54-39](https://user-images.githubusercontent.com/47572137/234034758-0b2e0e5b-3fa9-476a-b1d8-8047a0c26064.png)
请问使用vulkan-copy出现图中这种显示错误会是什么导致的？换回dxva-copy可以正常显示。
使用了以下设置：（使用mpv版本为[hingbong/mpv-winbuild](https://github.com/hingbong/mpv-winbuild/actions/runs/4691884289)）
```
vo = gpu-next
gpu-context = winvk
hwdec=vulkan-copy
```

**Reply by zxhzxhz** [reply]
> > ![Snipaste_2023-04-24_22-54-39](https://user-images.githubusercontent.com/47572137/234034758-0b2e0e5b-3fa9-476a-b1d8-8047a0c26064.png) 请问使用vulkan-copy出现图中这种显示错误会是什么导致的？换回dxva-copy可以正常显示。 使用了以下设置：（使用mpv版本为 [hingbong/mpv-winbuild ](https://github.com/hingbong/mpv-winbuild/actions/runs/4691884289)）
> > ```
> > vo = gpu-next
> > gpu-context = winvk
> > hwdec=vulkan-copy
> > ```
> 
> nvidia驱动的问题

刚刚试着更新到最新的gameready driver测试,还是同样的问题,看来暂时是用不上vulkan了

**Reply by hooke007** [reply]
vulkan解码支持的格式不多，，反过来写更省“笔墨”
